src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs

```diff
@@ -139,6 +142,28 @@ public FoldingRangeRegistrationOptions GetRegistrationOptions(FoldingRangeCapabi
             return new Container<FoldingRange>(mappedRanges);
         }
 
+        /// <summary>
+        /// Fixes the start of a range so that the offset of the first line is the last character on that line. This makes
+        /// it so collapsing will still show the text instead of just "..."
+        /// </summary>
+        private void FixRangeStart(Range mappedRange, RazorCodeDocument codeDocument)
+        {
+            Debug.Assert(mappedRange.Start.Line > mappedRange.End.Line);
+
+            var sourceText = codeDocument.GetSourceText();
+            var startLine = mappedRange.Start.Line;
+            var lineSpan = sourceText.Lines[startLine].Span;
+
+            // Search from the end of the line to the beginning for the first non whitespace character. We want that
```

Comment:
Do you need to be careful to not cross a language boundary? Or are there no foldable constructs that allow multiple languages on the same line?

Reply:
I'm not actually sure. Are you thinking something like: 

```
<div>@code {

}</div>
```
?

Reply:
Yeah, something like that.

Reply:
Looks like we do have to be careful. Should likely just provide one per line. 

![e54eebb4-3c8e-474b-9a9f-a3e158c68691](https://user-images.githubusercontent.com/475144/163290654-992878d0-3864-4dc3-b3af-1e28e3617817.gif)

Reply:
This is a great question. Ya you can def have `<div>@code {` can also have:
```HTML
<div>@if(true) { <p>
<p>
}
</div>
```

Reply:
ooh yeah, that looks like a bug no matter where the folding range goes :)